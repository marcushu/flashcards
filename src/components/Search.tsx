import { useState, useRef } from 'react';

interface SearchProps {
  setTopic: (topic: string) => void
}

const Search = ({ setTopic }: SearchProps) => {
  const [searchText, setsearchText] = useState("");
  const textRef = useRef<HTMLInputElement>(null);
  
  const sendTopic = () => {
    setTopic(searchText);
    setsearchText("");

    // kludge alert
    if(textRef && textRef.current) {
      textRef.current.value = "";
    }    
  }

  return (
    <div style={{ paddingTop: "40px" }}>
      <input type="text"
        placeholder="Topic Search..."
        ref={textRef}
        id="searchBox"
        onChange={e => setsearchText(e.currentTarget.value)} />
      <button onClick={sendTopic}>
        <svg
          xmlns="http://www.w3.org/2000/svg"
          width="20"
          height="20"
          fill="currentColor"
          className="bi bi-search"
          viewBox="0 0 16 16">
          <path fillRule="evenodd"
            d="M10.442 10.442a1 1 0 0 1 1.415 0l3.85 3.85a1 1 0 0 1-1.414 1.415l-3.85-3.85a1 1 0 0 1 0-1.415z" />
          <path fillRule="evenodd"
            d="M6.5 12a5.5 5.5 0 1 0 0-11 5.5 5.5 0 0 0 0 11zM13 6.5a6.5 6.5 0 1 1-13 0 6.5 6.5 0 0 1 13 0z" />
        </svg>
      </button>
      
    </div>
  )
}

export default Search