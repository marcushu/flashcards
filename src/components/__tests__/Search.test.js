import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Search from '../Search';

describe("Search", () => {
    const mockSetFn = jest.fn();

    it("renders", () => {
        const { asFragment } = render(
            <Search setTopic={mockSetFn} />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it("calls the set function with the correct text", () => {
        const { getByRole } = render(
            <Search setTopic={mockSetFn} />
        );

        fireEvent.change(getByRole('textbox'), { target: {value: "searchtext"}});
        fireEvent.click(getByRole('button'));

        expect(mockSetFn).toHaveBeenCalledWith("searchtext");
    });
})