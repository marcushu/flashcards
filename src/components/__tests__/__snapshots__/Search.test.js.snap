// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders 1`] = `
<DocumentFragment>
  <div
    style="padding-top: 40px;"
  >
    <p>
      Search for a topic.
    </p>
    <input
      id="searchBox"
      type="text"
    />
    <button>
      <svg
        class="bi bi-search"
        fill="currentColor"
        height="20"
        viewBox="0 0 16 16"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.442 10.442a1 1 0 0 1 1.415 0l3.85 3.85a1 1 0 0 1-1.414 1.415l-3.85-3.85a1 1 0 0 1 0-1.415z"
          fill-rule="evenodd"
        />
        <path
          d="M6.5 12a5.5 5.5 0 1 0 0-11 5.5 5.5 0 0 0 0 11zM13 6.5a6.5 6.5 0 1 1-13 0 6.5 6.5 0 0 1 13 0z"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
